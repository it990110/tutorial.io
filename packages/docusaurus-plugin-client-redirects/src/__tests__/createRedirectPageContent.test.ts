/**
 * Copyright (c) it990110, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import createRedirectPageContent from '../createRedirectPageContent';

describe('createRedirectPageContent', () => {
  it('works', () => {
    expect(
      createRedirectPageContent({toUrl: 'https://tutorial.io/'}),
    ).toMatchSnapshot();
  });

  it('encodes uri special chars', () => {
    const result = createRedirectPageContent({
      toUrl: 'https://tutorial.io/gr/σελιδας/',
    });
    expect(result).toContain(
      'https://tutorial.io/gr/%CF%83%CE%B5%CE%BB%CE%B9%CE%B4%CE%B1%CF%82/',
    );
    expect(result).toMatchSnapshot();
  });
});
