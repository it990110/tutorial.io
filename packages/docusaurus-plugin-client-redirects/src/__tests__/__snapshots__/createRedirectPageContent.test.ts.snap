// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createRedirectPageContent encodes uri special chars 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <meta http-equiv="refresh" content="0; url=https://tutorial.io/gr/%CF%83%CE%B5%CE%BB%CE%B9%CE%B4%CE%B1%CF%82/">
    <link rel="canonical" href="https://tutorial.io/gr/%CF%83%CE%B5%CE%BB%CE%B9%CE%B4%CE%B1%CF%82/" />
  </head>
  <script>
    window.location.href = 'https://tutorial.io/gr/%CF%83%CE%B5%CE%BB%CE%B9%CE%B4%CE%B1%CF%82/' + window.location.search + window.location.hash;
  </script>
</html>"
`;

exports[`createRedirectPageContent works 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <meta http-equiv="refresh" content="0; url=https://tutorial.io/">
    <link rel="canonical" href="https://tutorial.io/" />
  </head>
  <script>
    window.location.href = 'https://tutorial.io/' + window.location.search + window.location.hash;
  </script>
</html>"
`;
